src/core/components/auth/authorize-operation-btn.jsx

```diff
@@ -17,17 +18,18 @@ export default class AuthorizeOperationBtn extends React.Component {
   }
 
   render() {
-    let { isAuthorized } = this.props
+    let { describedBy, isAuthorized } = this.props
 
     return (
       <button className={isAuthorized ? "authorization__btn locked" : "authorization__btn unlocked"}
-        aria-label={isAuthorized ? "authorization button locked" : "authorization button unlocked"}
+        aria-describedBy={describedBy ? describedBy : null}
+        aria-haspopup="dialog"
+        aria-label={isAuthorized ? "authorize locked" : "authorize unlocked"}
```

Comment:
Admittedly I find the UX of the authorisation stuff a bit unclear but as I understand it we get the "Locked" state when you have put in authentication credentials that the route requires. is that right?

If so, I wonder whether locked/unlocked is the right thing to read out here? Since the thing you’re checking is whether it is authorised I wonder whether just "Authorised"/"Unauthorised" would be better labels for this?

Having said that I might just be misunderstanding the logic for these padlock states

Reply:
I'm actually not sure -- I was half-hoping someone more familiar with Swagger UI would comment and suggest better wording. Absent that, I just went with locked/unlocked since it was used before and it gave the closest parity with the visual experience.

Reply:
maybe based on the actions in the popup, "Authorize" and "Remove authorization" might fit best?

Reply:
Would be good to get some feedback here @shockey and @webron. What would be the clearer description for what these values mean given you might not have the visual context of the entire page?

Reply:
> @leggsimon: Having said that I might just be misunderstanding the logic for these padlock states

You wouldn't be the first - just take a look at #4402! We're moving away from the lock metaphor in general, it's clear it has _not_ been a hit with users.

I can't share the new design proposal publicly yet, but suffice it to say that it unambiguously communicates two states: "authorization is available for this request, `and has`/`but has not` been provided".

I see no reason to torture screen reader users with a visual metaphor that even visual users don't like!

---

**Change request**: let's change this label! 

@smhigley, when we last chatted I don't think we quite landed on an decision for what `aria-label` should represent in the context of a button that both conveys current state and triggers a dialog that allows users to mutate state - should the label describe the current state or what the button allows the user to change?

If we're focusing on current state, I think something like `Authorization is active` and `Authorization is available but not active` makes sense, but we can be more terse if that's conventional for labels, e.g. `Authorization active` and `Authorization available`.

Conversely, if we're focusing on mutations that the user can trigger, your suggestion of `Authorize` and `Remove authorization` is good.

Which road to take is up to you, @smhigley!